crates/biome_js_analyze/tests/specs/correctness/noUnusedVariables/invalidRestSiblings.js.snap

```diff
@@ -0,0 +1,87 @@
+---
+source: crates/biome_js_analyze/tests/spec_tests.rs
+assertion_line: 96
+expression: invalidRestSiblings.js
+---
+# Input
+```js
+// With ignoreRestSiblings: false, unused variables should be reported
+const car = { brand: "Tesla", year: 2019, countryCode: "US" };
+const { brand, year, ...other } = car;
+console.log(other);
+
+// Renamed properties
+const data = { foo: 1, bar: 2, baz: 3 };
+const { foo: renamedFoo, bar: renamedBar, ...remaining } = data;
+console.log(remaining);
+
+```
+
+# Diagnostics
+```
+invalidRestSiblings.js:3:9 lint/correctness/noUnusedVariables ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! This variable is unused.
+  
+    1 │ // With ignoreRestSiblings: false, unused variables should be reported
+    2 │ const car = { brand: "Tesla", year: 2019, countryCode: "US" };
+  > 3 │ const { brand, year, ...other } = car;
+      │         ^^^^^
+    4 │ console.log(other);
+    5 │ 
+  
+  i Unused variables usually are result of incomplete refactoring, typos and other source of bugs.
+  
```

Comment:
I wonder if we could add a note indicating that users can ignore rest sibling with the `ignoreRestSiblings` option.
We could display this note only for rest siblings like in the current case.

Reply:
Great idea! I added that as a separate note, but let me know if you’d like to have them merged into one.